common/src/app/common/files/validate.cljc

```diff
@@ -393,7 +393,8 @@
                   (check-shape-copy-root-top shape file page libraries)))
 
               (if (ctk/main-instance? shape)
-                (if (= context :not-component)
+                ;; mains can't be nested into mains
+                (if (or (= context :not-component) (= context :main-top))
```

Comment:
I think that this case should always report error. If you reach here, this is a main instance and it's not root, so it must be forbidden.

The `check-shape-main-root-nested` function is no longer relevant (better not to delete it, but now it won't be called).

Reply:
I think it's more resilient this way, we can have generated this situation in some files on PRE environments, this way those files can be fixed automatically too

Reply:
If we have a file has any incorrect situation because it was edited before this change, the validate function must report it as error, precisely to be able to get it repaired, doesn't it?

Reply:
That's what we are doing, reporting this situation as an error

Reply:
But we need to talk about a strategy to repair files in PRE when this is deployed, to not break the work of the designers.